optuna/visualization.py

```diff
@@ -0,0 +1,70 @@
+from optuna.structs import TrialState
+from optuna.study import Study  # NOQA
+from typing import List  # NOQA
+
+try:
+    import plotly.graph_objs as go
+    from plotly.offline import init_notebook_mode
+    from plotly.offline import iplot
+    _available = True
+except ImportError as e:
+    _import_error = e
+    # Visualization features are disabled because plotly is not available.
+    _available = False
+
+
+def plot_intermediate_values(study):
```

Comment:
I'm unfamiliar with `plotly` and Jupyter Notebook, so it would be very helpful if you could upload an example plotted image to this PR.

Reply:
I added a notebook to demonstrate the function: https://github.com/pfnet/optuna/blob/visualize-learning-curve/examples/visualization/plot_intermediate_values.ipynb